src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/TestProgressStateAwareTerminal.cs

```diff
@@ -135,9 +135,12 @@ internal void WriteToTerminal(Action<ITerminal> write)
         }
         else
         {
-            _terminal.StartUpdate();
-            write(_terminal);
-            _terminal.StopUpdate();
+            lock (_lock)
```

Comment:
FYI https://learn.microsoft.com/en-us/dotnet/api/system.threading.lock?view=net-9.0 but it's there starting from 9